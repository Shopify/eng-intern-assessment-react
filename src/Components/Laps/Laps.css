.table-container{
    margin: 0 auto;
    width: 98%;

    th,td{
        text-align: center;
        color: #333;
    }

    th{
        font-size: 1.4rem;
    }

    td{
        font-size: 1.2rem;
    }

    
}