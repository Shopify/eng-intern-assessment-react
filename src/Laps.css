.laps-container {
    margin: 3rem 0;
    width: 100%;
}

.laps-title {
    display: flex;
    justify-content: space-between;
    font-size: 1.2rem;
    font-weight: bold;
    margin-bottom: 1.1rem;
}

.lap-row {
    display: flex;
    justify-content: space-between;
}

.lap-column {
    flex: 1;
    text-align: center;
}