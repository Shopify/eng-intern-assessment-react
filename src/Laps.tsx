import React from "react";
import { formattedTime } from "./utils";

type LapsProps = {
  lapTimes: number[];
};

export default function Laps({ lapTimes }: LapsProps) {
  return (
    <div className="laps-display">
      {lapTimes.map((lapTime, index) => (
        <div className="lap-row" key={index}>
          <span className="lap-title">{`Lap ${index + 1}`}</span>
          <span className="lap-time">{formattedTime(lapTime)}</span>
        </div>
      ))}
    </div>
  );
}
