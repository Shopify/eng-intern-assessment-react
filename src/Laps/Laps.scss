@import './../variables.scss';

.lap-list {
  max-height: 700px;
  overflow: auto;
  display: flex;
  flex-direction: column-reverse;
  margin-top: 20px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  :nth-child(even) div {
    background-color: $primary;
  }
}

.lap-row {
  display: flex;
  font-size: 40px;
  border: 1px solid $secondary;
  div{
    width: 50%;
    display: flex;
    justify-content: center; /* For horizontal alignment */
  }
  :last-child{
    border-left: 1px solid $secondary;
  }
}