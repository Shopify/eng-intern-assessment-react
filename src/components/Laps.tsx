import React from "react";
import formatTime from "../utils/formatTime";

interface Lap {
  number: number;
  totalTime: number;
  timeInterval: number;
}

interface LapsProps {
  laps: Lap[];
}

const Laps = ({ laps }: LapsProps) => {
  return (
    <>
      <div className="laps__title">
        <h3>Laps</h3>
        <div className="lap__times">
          <span className="lap__times__time">Time</span>
          <span className="lap__times__time">Interval</span>
        </div>
      </div>
      <div className="laps-list" data-testid="laps-list">
        {laps.map((lap) => {
          return (
            <div key={lap.number} className="lap">
              <span>{lap.number}</span>
              <div className="lap__times">
                <span className="lap__times__time">
                  {formatTime(lap.totalTime)}
                </span>
                <span className="lap__times__time">
                  {formatTime(lap.timeInterval)}
                </span>
              </div>
            </div>
          );
        })}
      </div>
    </>
  );
};

export default Laps;
