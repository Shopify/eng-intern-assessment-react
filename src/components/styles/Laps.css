ul{
    font-family: monospace;
    list-style: none;
    margin: 0;
    padding: 0;
}

.lap-table-container{
    width: 100%;
}

.laps-list {
    display: flex;
    flex-direction: column-reverse;
    
}

.scrollable-table{
    max-height: 300px;
    overflow-y: scroll;
    scrollbar-gutter: stable;
}

.lap-listing{
    margin-right: -2rem;
    border-top: 1px solid #4d6883ab;
    font-family: 'Roboto Mono' !important;
    font-size: 1.2rem;
    font-weight: 500;
    display: flex;
    flex-direction: row;
    list-style-type: none;
    justify-content: space-between;
    margin-left: 1rem;
    margin-right: 1rem;
    padding: 4px 8px;
}

.current-lap{
    border-top: none;
    background-color: #74b4f123;
    border-radius: 4px;
    border-bottom-left-radius: 0px;
    border-bottom-right-radius: 0px;
}


::-webkit-scrollbar {
    width: 5px;
    height: 5px;
  }
  
  ::-webkit-scrollbar-track {
    background:rgba(100, 100, 100, 0.627) ;
    /* -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3); */
    /* -webkit-border-radius: 10px; */
    /* border-radius: 10px; */
  }
  
  ::-webkit-scrollbar-thumb {
    -webkit-border-radius: 10px;
    border-radius: 10px;
    background: rgb(15, 24, 33);
  }
  
  ::-webkit-scrollbar-thumb:window-inactive {
    background: rgba(255, 255, 255, 0.101);
  }