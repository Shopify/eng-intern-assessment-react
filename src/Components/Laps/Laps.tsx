import React from 'react'
import Table from 'react-bootstrap/Table';
import Button from 'react-bootstrap/Button';
import './Laps.css'

/*
  Define the props for the Laps component
  lapTimes: an array of strings representing the lap times
  onDelete: a function that takes an index and deletes the lap time at that index
*/

interface LapsProps {
    lapTimes: string[];
    onDelete: (index: number) => void;
  }


/*
  Lap component that displays the lap times in a table
*/

export default function Laps({lapTimes, onDelete}: LapsProps) {
    return(
        <div className="table-container" data-testid="lap-list">
          <Table bordered hover>
          <thead>
            <tr>
              <th>Lap #</th>
              <th>Time</th>
              <th>Action</th>
            </tr>
          </thead>
          <tbody>
            {lapTimes.map((lapTime: string, index: number) => (
              <tr key={index}>
                <td>{index + 1}</td>
                <td data-testid={`lap-time-${index}`}>
                  {lapTime}
                </td>
                <td>
                    <Button variant="dark" size="sm" onClick={() => onDelete(index)}>
                      Delete
                    </Button>
                  </td>
              </tr>
            ))}
          </tbody>
        </Table>
      </div>
    )
}