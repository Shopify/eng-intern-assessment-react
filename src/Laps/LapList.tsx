import {formatTime} from "../utils";
import React from "react";
import {Lap} from "./types";
import './Laps.scss'
const LapList = ({laps}: { laps: Array<Lap> }) => (
        <div className="lap-list">
            {laps.map((lap) => (
                <div className="lap-row" key={lap.lapNumber}>
                    <div>Lap: {lap.lapNumber}</div>
                    <div>{formatTime(lap.lapTime)}</div>
                </div>
            ))}
        </div>
    )


export default LapList