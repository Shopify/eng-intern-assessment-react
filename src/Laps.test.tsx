import React from 'react';
import renderer from 'react-test-renderer';
import Laps from './Laps';

describe('Laps tests', () => {
    it('render should match snapshot file with empty recordedLapTimes array', () => {
        const tree = renderer.create(<Laps recordedLapTimes={[]}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('render should match snapshot file with non-empty recordedLapTimes array', () =>{
        const tree = renderer.create(<Laps recordedLapTimes={[1000,10000,20000,25000,30000]}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})