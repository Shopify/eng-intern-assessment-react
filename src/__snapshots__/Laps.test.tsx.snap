// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Laps tests render should match snapshot file with empty recordedLapTimes array 1`] = `null`;

exports[`Laps tests render should match snapshot file with non-empty recordedLapTimes array 1`] = `
<div>
  Lap : Time
  <ol>
    <li>
      00:00:01.00
    </li>
    <li>
      00:00:09.00
    </li>
    <li>
      00:00:10.00
    </li>
    <li>
      00:00:05.00
    </li>
    <li>
      00:00:05.00
    </li>
  </ol>
</div>
`;
